README.md

```diff
@@ -56,10 +61,41 @@ op.attempt(next => {
 
 ```
 
+### `Async/await` Support
+
+Retryme now supports `async/await`. It accepts three arguments:
+
+- `retry`: The configuration for create a new [retryme instance][retryme-instance].
+- `attempt`: The real `async` function which will be retried with `await`.
+- `ignore`: The `ignore` function also for setup [retryme instance][retryme-instance] to terminate retries.
+
+A simple example of how to use it is as follows:
+
+```js
+const awaitRetryme = require('retryme/await');
+
+function some_async_func(val) {
+  return then: (resolve, reject) => {
+    if (some_condition) return reject(some_error);
+    resolve(val);
+  }
+}
+
+// example of pass-in params.
+async function top-caller() {
+  await awaitRetryme({
+    retries: 3,
+    min: 5,
+    max: 10
+  }, some_async_func(some_value), some_ignore_function_if_exists);
```

Comment:
Im realizing that the reason this works is only because its a thenable. If this were an actual `async` function or a function that returns a promise, it would not work because you cannot `await` it more than once. So in this case we actually have to assume its an async function (that hasn't been executed like we are doing here) so that in our retry machinery we are executing the function. `await some_async_func()` 

its a similar convention taken in https://github.com/sindresorhus/p-retry and in https://github.com/sindresorhus/p-limit since promises by definition are "one and done".

Reply:
Leave a note here for note. Yea we designed the functionality when we only used in `ui-test-service` and we have thenables everywhere which makes it work as expected. But yes it cannot work well with `Promise`. Will try to find a way to refactor it. =)